test/unit/host_test.rb

```diff
@@ -2137,6 +2137,13 @@ def to_managed!
     assert_equal 'foo', host.shortname
   end
 
+  test 'check operatingsystem and architecture association' do
+    primary = FactoryGirl.build(:nic_managed, :primary => true, :name => 'foo')
+    host = FactoryGirl.build(:host, :interfaces => [primary, FactoryGirl.build(:nic_managed, :provision => true)])
+    host.populate_fields_from_facts(:architecture => "x86_64", :operatingsystem => 'RedHat', :operatingsystemrelease => '6.2')
+    assert host.operatingsystem.architectures.include?(host.architecture), "no association between operatingsystem and architecture"
```

Comment:
I added a check  to make sure that the operating system is nil before the populate_fields_from_facts call, couldn't use assert_difference in this case because the populate_fields_from_facts call creates the operating system hence i wont be able to check it before and see a change has occurred.